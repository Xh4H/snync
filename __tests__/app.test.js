const fs = require('fs')
const path = require('path')
const os = require('os')
const childProcess = require('child_process')
const rimraf = require('rimraf')
const { testProject } = require('../src')

let tmpDir

beforeAll(() => {
  tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'snync-'))
  childProcess.spawnSync('git', ['clone', 'git@github.com:snyk-labs/snync-fixtures.git'], {
    cwd: tmpDir
  })
})

afterAll(() => {
  rimraf.sync(tmpDir)
})

test('integration test', async () => {
  let out = ''
  await testProject(`${tmpDir}/snync-fixtures`, (...args) => (out += `${args.join(' ')}\n`))
  expect(out).toMatchSnapshot()
})
