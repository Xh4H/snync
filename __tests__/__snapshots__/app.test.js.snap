// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration test 1`] = `
"
Reviewing your dependencies...

Checking dependency: asunc
  ->  ❌ suspicious
Checking dependency: a-package-in-private-registry-0
  ->  ⚠️ vulnerable
Checking dependency: lodash
"
`;
